operations.js

```diff
@@ -0,0 +1,223 @@
+const utils = require('./utils');
+const db = require('@splinterlands/pg-querybuilder');
+const interface = require('@splinterlands/hive-interface');
+
+let hive = null;
+
+let _options = {
+	logging_level: 3,
+	rpc_error_limit: 20,
+	rpc_nodes: ["https://api.hive.blog", "https://anyx.io", "https://hived.splinterlands.com"],
+	ws_url: "http://localhost:1234",
+	state_file_name: 'sl-state.json',
+	game_api_url: 'http://localhost:3000',
+	prefix: "dev-sm_"
+};
+
+function init(options) {
+	_options = Object.assign(_options, options);
+	utils.set_options(_options);
+	hive = new interface.Hive(_options);
+
+	if(_options.connection)
+		db.init({ connection: _options.connection });
+}
+
+async function processPurchase(purchase_id, payment) {
+	return new Promise(async (resolve, reject) => {
+		let purchase = await db.lookupSingle('purchases', { uid: purchase_id });
```

Comment:
lookupSingle is dangerous as it doesn't not error when db has issues. I'm working on a replacement for it called lookupSingleOrError which will properly error. Consider incorporating this in the future.

Reply:
Also note that the db class throws errors.

Reply:
(My comments are for the Steemmonsters repo and pretty sure the db class was a copy paste of it.)

Reply:
Not part of this change!